crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -132,6 +178,61 @@ pub fn calculate_bounds(
     }
 }
 
+/// Updates [`Aabb`]s for [`Entities`](Entity) with [`Mesh`]es. This includes `Entities` that have
+/// been assigned new `Mesh`es as well as `Entities` whose `Mesh` has been directly mutated.
+///
+/// To opt out of bound calculation for an `Entity`, give it the [`NoFrustumCulling`] component.
+pub fn update_bounds(
+    mut commands: Commands,
+    meshes: Res<Assets<Mesh>>,
+    mut mesh_reassigned: Query<
+        (Entity, &Handle<Mesh>, &mut Aabb),
+        (Changed<Handle<Mesh>>, Without<NoFrustumCulling>),
+    >,
+    mut entity_mesh_rel: ResMut<EntityMeshRelationships>,
+    mut mesh_events: EventReader<AssetEvent<Mesh>>,
+    entities_lost_mesh: RemovedComponents<Handle<Mesh>>,
+) {
+    // If an entity's mesh handle was removed, unlink them.
+    for entity in entities_lost_mesh.iter() {
+        entity_mesh_rel.deregister(entity);
+    }
+
+    // If an entity with an Aabb was assigned a new mesh, move it to the new list and update its
+    // Aabb.
+    for (entity, mesh_handle, mut aabb) in mesh_reassigned.iter_mut() {
+        // Remove it from the previous list
+        entity_mesh_rel.deregister(entity);
+
+        // Add it to the new list.
+        entity_mesh_rel.register(entity, mesh_handle);
+
+        // Update its Aabb
+        if let Some(mesh) = meshes.get(mesh_handle) {
+            if let Some(new_aabb) = mesh.compute_aabb() {
+                *aabb = new_aabb;
+            }
+        }
+    }
```

Comment:
```suggestion
    for (entity, mesh_handle, mut aabb) in mesh_reassigned.iter_mut() {
        entity_mesh_rel.deregister(entity);
        entity_mesh_rel.register(entity, mesh_handle);
        if let Some(mesh) = meshes.get(mesh_handle) {
            if let Some(new_aabb) = mesh.compute_aabb() {
                *aabb = new_aabb;
            }
        }
    }
```

Please don't use comments that describe **what** the code does. You should only write comments if it's unclear **why** it does something. The code is **what** the code does, comments that tell **what** the code does are redundant.

Reply:
Ah, good catch. I 100% agree.

Reply:
Comments removed in 76197ccee336f6f073038959b11d19f740333243.